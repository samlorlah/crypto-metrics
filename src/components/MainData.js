import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { fetchCoins } from '../redux/coinList/coinList';
import styles from './MainData.module.css';
import MainDataItem from './MainDataItem';

const MainData = () => {
  const { coins } = useSelector((state) => state.coins);
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(fetchCoins());
  }, []);
  return (
    <div className={styles.dataContainer}>
      <ul>
        {coins.map((coin) => (
          <MainDataItem
            key={coin.id}
            id={coin.id}
            symbol={coin.symbol}
            coinName={coin.coinName}
            price={coin.price}
          />
        ))}
      </ul>
    </div>
  );
};

export default MainData;
