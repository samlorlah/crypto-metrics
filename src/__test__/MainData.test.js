import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import MainData from '../components/MainData';

describe('Test Compenent that renders all the lists of coins', () => {
  it('get the list of coins from store and display', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <MainData />
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });
});