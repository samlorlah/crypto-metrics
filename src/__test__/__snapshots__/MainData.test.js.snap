// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Compenent that renders all the lists of coins get the list of coins from store and display 1`] = `
<div
  className="dataContainer"
>
  <ul />
</div>
`;
